src/System.Management.Automation/engine/parser/ast.cs

```diff
@@ -5342,6 +5342,215 @@ internal override AstVisitAction InternalVisit(AstVisitor visitor)
+        /// </summary>
+        /// <param name="lhsStatement">The statement to the left of the operator.</param>
+        /// <param name="rhsStatement">The statement to the right of the operator.</param>
+        /// <param name="chainOperator">The operator used.</param>
+        /// <param name="extent">The extent of the chained statement.</param>
+        public StatementChainAst(
+            IScriptExtent extent,
+            PipelineBaseAst lhsStatement,
+            StatementAst rhsStatement,
+            TokenKind chainOperator)
+            : base(extent)
+        {
+            if (lhsStatement == null)
+            {
+                throw new ArgumentNullException(nameof(lhsStatement));
+            }
+
+            if (rhsStatement == null)
+            {
+                throw new ArgumentNullException(nameof(rhsStatement));
+            }
+
+            if (chainOperator != TokenKind.AndAnd && chainOperator != TokenKind.OrOr)
+            {
+                throw new ArgumentException(nameof(chainOperator));
+            }
+
+            LhsStatement = lhsStatement;
+            RhsStatement = rhsStatement;
+            Operator = chainOperator;
+
+            SetParent(LhsStatement);
+            SetParent(RhsStatement);
+        }
+
+        /// <summary>
+        /// The left hand statement in the chain.
+        /// </summary>
+        public PipelineBaseAst LhsStatement { get; }
+
+        /// <summary>
+        /// The right hand statement in the chain.
+        /// </summary>
+        public StatementAst RhsStatement { get; }
+
+        /// <summary>
+        /// The chaining operator used.
+        /// </summary>
+        public TokenKind Operator { get; }
+
+        /// <summary>
+        /// Create a copy of this Ast.
+        /// </summary>
+        public override Ast Copy()
+        {
+            return new StatementChainAst(Extent, CopyElement(LhsStatement), CopyElement(RhsStatement), Operator);
+        }
+
+        internal override object Accept(ICustomAstVisitor visitor)
+        {
+            if (!(visitor is ICustomAstVisitor2 visitor2))
+            {
+                return null;
+            }
+
+            return visitor2.VisitStatementChain(this);
+        }
+
+        internal override AstVisitAction InternalVisit(AstVisitor visitor)
+        {
+            var visitAction = AstVisitAction.Continue;
+            if (visitor is AstVisitor2 visitor2)
+            {
+                visitAction = visitor2.VisitStatementChain(this);
+
+                if (visitAction != AstVisitAction.Continue)
+                {
+                    return visitAction;
+                }
+            }
+
+            visitAction = LhsStatement.InternalVisit(visitor);
+
+            if (visitAction == AstVisitAction.StopVisit)
+            {
+                return visitAction;
+            }
+
+            return RhsStatement.InternalVisit(visitor);
```

Comment:
You need to run `CheckForPostAction` after visiting this node. Also, the way you check `visitAction` is not right. Please check how `BinaryExpressionAst` does it and you probably can follow the same.